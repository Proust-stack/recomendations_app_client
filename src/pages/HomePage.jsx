import React, { Suspense, useEffect } from "react";
import { Box } from "@mui/material";
import Grid from "@mui/material/Grid";
import { useDispatch, useSelector } from "react-redux";
import { ErrorBoundary } from "react-error-boundary";
import styled from "@mui/material/styles/styled";
import { useTheme } from "@mui/material/styles";
import useMediaQuery from "@mui/material/useMediaQuery";

import { getAllReviewsByTags } from "../slices/reviewSlice";
import TagsCloud from "../components/TagsCloud";
import ErrorFallback from "../utils/errorCallback";
import Loader from "../components/ui/Loader";
import LazyShortReviewCardComponent from "../hocs/LazyComponent";

const StyledBox = styled(Box)(({ theme }) => ({
  display: "flex",
  alignItems: "flex-start",
  justifyContent: "flex-start",
  flexDirection: "row",
  flexWrap: "wrap",
  gap: 10,
}));

export default function HomePage() {
  const theme = useTheme();
  const matches = useMediaQuery(theme.breakpoints.down("md"));
  const dispatch = useDispatch();
  const { reviewsAll } = useSelector((state) => state.review);
  const { selectedTags } = useSelector((state) => state.tag);

  useEffect(() => {
    dispatch(getAllReviewsByTags(selectedTags));
  }, [selectedTags]);

  return (
    <ErrorBoundary FallbackComponent={ErrorFallback}>
      <Suspense fallback={<Loader />}>
        <Grid
          container
          spacing={2}
          sx={{ padding: 2 }}
          direction={matches ? "column" : "row"}
        >
          <Grid item xs>
            <TagsCloud />
          </Grid>
          <Grid item xs={10}>
            <Box
              sx={{
                display: "flex",
                alignItems: "center",
                justifyContent: "center",
                flexDirection: "row",
                flexWrap: "wrap",
                gap: 2,
                padding: 1,
              }}
            >
              {reviewsAll &&
                reviewsAll.map((review) => (
                  <StyledBox key={review._id}>
                    <LazyShortReviewCardComponent {...review} />
                  </StyledBox>
                ))}
            </Box>
          </Grid>
        </Grid>
      </Suspense>
    </ErrorBoundary>
  );
}
