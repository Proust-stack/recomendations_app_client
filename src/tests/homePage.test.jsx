import { JoinLeftSharp } from "@mui/icons-material";
import { cleanup, fireEvent, screen, render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import * as reduxHooks from "react-redux";

import HomePage from "../pages/HomePage";

jest.mock("react-redux");

describe("HomePage", () => {
  //afterEach(cleanup);
  it("page snapshot", () => {
    const view = render(<HomePage />);
    expect(view).toMatchSnapshot();
  });
  it("render tags", () => {
    jest.spyOn(reduxHooks, "useSelector").mockReturnValue([]);
    //expect(view).toMatchSnapshot();
  });
});
