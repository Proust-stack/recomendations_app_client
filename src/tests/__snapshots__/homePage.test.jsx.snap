// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HomePage page snapshot 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1iu77a7-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
        >
          <div
            class="MuiBox-root css-z1n926"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 MuiGrid-grid-xs-10 css-1korw7z-MuiGrid-root"
        >
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1an47wb-MuiTypography-root"
            >
              category_lattest
            </h6>
            <div
              class="MuiBox-root css-habezc"
            />
          </div>
          <div
            class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
          >
            <h6
              class="MuiTypography-root MuiTypography-subtitle2 css-1an47wb-MuiTypography-root"
            >
              category_hottest
            </h6>
            <div
              class="MuiBox-root css-habezc"
            />
          </div>
        </div>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="MuiGrid-root MuiGrid-container MuiGrid-spacing-xs-2 css-1iu77a7-MuiGrid-root"
    >
      <div
        class="MuiGrid-root MuiGrid-item MuiGrid-grid-xs-true css-1vd824g-MuiGrid-root"
      >
        <div
          class="MuiBox-root css-z1n926"
        />
      </div>
      <div
        class="MuiGrid-root MuiGrid-container MuiGrid-item MuiGrid-spacing-xs-4 MuiGrid-grid-xs-10 css-1korw7z-MuiGrid-root"
      >
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1an47wb-MuiTypography-root"
          >
            category_lattest
          </h6>
          <div
            class="MuiBox-root css-habezc"
          />
        </div>
        <div
          class="MuiGrid-root MuiGrid-item css-13i4rnv-MuiGrid-root"
        >
          <h6
            class="MuiTypography-root MuiTypography-subtitle2 css-1an47wb-MuiTypography-root"
          >
            category_hottest
          </h6>
          <div
            class="MuiBox-root css-habezc"
          />
        </div>
      </div>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
